File: Game2/server/Game/Buf/BufExtra.cs
using System;
using System.Collections.Generic;
using System.Text;
using Game.Buf;

namespace Game.Buf
{
    public class BufExtra : Buf
    {
        private BBufExtra extra;

        public BufExtra(BBuf bean, BBufExtra extra) : base(bean)
        {
            this.extra = extra;
        }

        public override void CalculateFighter(Game.Fight.Fighter fighter)
        {
            fighter.Bean.Attack += 10.0f;
            fighter.Bean.Defence += 10.0f;
        }
    }
}

File: ZezexJava/server/src/Game/Buf/BufExtra.java
package Game.Buf;

import Game.Buf.*;
import Game.*;

public class BufExtra extends Buf {
	private BBufExtra extra;

	public BufExtra(BBuf bean, BBufExtra extra) {
		super(bean);
		this.extra = extra;
	}

	@Override
	public void CalculateFighter(Game.Fight.Fighter fighter) {
		fighter.getBean().Attack += 10.0f;
		fighter.getBean().Defence += 10.0f;
	}
}